x-pack/plugins/ml/public/application/timeseriesexplorer/timeseriesexplorer_embeddable_chart/timeseriesexplorer_title.tsx

```diff
@@ -0,0 +1,152 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import type { FC } from 'react';
+import React from 'react';
+import {
+  EuiButtonIcon,
+  EuiFlexGroup,
+  EuiFlexItem,
+  EuiTitle,
+  EuiTextColor,
+  EuiToolTip,
+} from '@elastic/eui';
+import { i18n } from '@kbn/i18n';
+import { FormattedMessage } from '@kbn/i18n-react';
+import { type MlEntityFieldOperation, ML_ENTITY_FIELD_OPERATIONS } from '@kbn/ml-anomaly-utils';
+
+import { useMlKibana } from '../../contexts/kibana';
+import type { MlEntity, SingleMetricViewerEmbeddableApi } from '../../../embeddables/types';
+import { TimeSeriesExplorerHelpPopover } from '../timeseriesexplorer_help_popover';
+import { SINGLE_METRIC_VIEWER_ENTITY_FIELD_SELECTION_TRIGGER } from '../../../ui_actions/triggers';
+
+const FilterButton: FC<{
+  api: SingleMetricViewerEmbeddableApi;
+  entity: MlEntity;
+  operation: MlEntityFieldOperation;
+}> = ({ api, entity, operation }) => {
+  const {
+    services: { uiActions },
+  } = useMlKibana();
+  const isAddFilter = operation === ML_ENTITY_FIELD_OPERATIONS.ADD;
+  return (
+    <EuiToolTip
+      content={
+        isAddFilter ? (
+          <FormattedMessage
+            id="xpack.ml.timeSeriesExplorer.title.addFilterTooltip"
+            defaultMessage="Filter for"
+          />
+        ) : (
+          <FormattedMessage
+            id="xpack.ml.timeSeriesExplorer.title.addNegateFilterTooltip"
+            defaultMessage="Filter out"
+          />
+        )
+      }
+    >
+      <EuiButtonIcon
+        size="xs"
+        onClick={() => {
+          const trigger = uiActions.getTrigger(SINGLE_METRIC_VIEWER_ENTITY_FIELD_SELECTION_TRIGGER);
+          trigger.exec({
+            embeddable: api,
+            data: [
+              {
+                ...entity,
+                operation: isAddFilter
+                  ? ML_ENTITY_FIELD_OPERATIONS.ADD
+                  : ML_ENTITY_FIELD_OPERATIONS.REMOVE,
+              },
+            ],
+          });
+        }}
+        iconType={isAddFilter ? 'plusInCircle' : 'minusInCircle'}
+        aria-label={
+          isAddFilter
+            ? i18n.translate('xpack.ml.timeSeriesExplorer.title.addFilterAriaLabel', {
+                defaultMessage: 'Filter for',
+              })
+            : i18n.translate('xpack.ml.timeSeriesExplorer.title.addNegateFilterAriaLabel', {
+                defaultMessage: 'Filter out',
+              })
+        }
+      />
+    </EuiToolTip>
+  );
+};
+
+interface SingleMetricViewerTitleProps {
+  api: any;
```

Comment:
can this `any` be replaced with the correct type?

Reply:
Updated in 41c37ae127f9a44b4f83250e2b3fe1def6953a11